### ISSUE TITLE
Struct empty time.Time, pq.NullTime and sql.NullString marshal give me not empty values

### ISSUE THREAD

**Issue by Dionid** [mailru/easyjson#141]
I have this struct:

```
type User struct {
        Id                  int64               `json:",omitempty" sql:"index"`
        Email            string              `sql:"index"`
        CreatedAt    time.Time       `json:",omitempty" db:"created_at"`
        DeletedAt    pq.NullTime    `json:",omitempty" db:"deleted_at"`
	Phone          sql.NullString  `json:",omitempty"`
}
```

SQL is:

```
err := db.Select(&users, `
	SELECT email, id FROM users
`)
```

Why am I getting this JSON with not empty values (phone, created_at, deleted_at) when I'm using MarshalJSON:

```
[
        {
                "email": "diodos@yandex.ru",
		"id": 1,
		"phone": {},
		"created_at": "0001-01-01T00:00:00Z",
		"deleted_at": {
			"time": "0001-01-01T00:00:00Z"
		}
	},
]
```

**Comment by Dionid** [comment]
I've found the answer!
I need to use pointers in User struct  like this (look at CreatedAt, DeletedAt, Phone):

```
type User struct {
	Id        int64        `json:",omitempty" sql:"index"`
	CreatedAt *time.Time   `json:",omitempty" db:"created_at"`
	DeletedAt *pq.NullTime `json:",omitempty" db:"deleted_at"`
	Email    string          `sql:"index"`
	Password string          `json:",omitempty"`
	IsAdmin  bool            `json:",omitempty" db:"is_admin"`
	Phone    *sql.NullString `json:",omitempty"`
}
```

**Comment by Dionid** [comment]
Even better:

```
type User struct {
	Id        int64        `json:",omitempty" sql:"index"`
	CreatedAt *time.Time   `json:",omitempty" db:"created_at"`
	DeletedAt *time.Time `json:",omitempty" db:"deleted_at"`
	Email    string          `sql:"index"`
	Password string          `json:",omitempty"`
	IsAdmin  bool            `json:",omitempty" db:"is_admin"`
	Phone    *string `json:",omitempty"`
}
```

**Comment by rvasily** [comment]
Need check encoding.TextMarshaler for time.Time
for pq.NullTime not sure, may be we cant do nothing here - no common agreement about structy emptyness

**Comment by GoWebProd** [comment]
Will fix in #28
